### PR TITLE
tool with proposed folder structure

### PR DESCRIPTION
**Pull request by Yiakman**

### CODE REVIEW

File: tools/smart-search/metadata.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        "description": "The sources used to generate the answer",
+        "items": {
+          "$ref": "#/$defs/SmartSearchSourcePage"
+        },
+        "type": "array"
+      },
+      "statements": {
+        "description": "The statements extracted from the sources",
+        "items": {
+          "$ref": "#/$defs/SmartSearchStatement"
```
Comment (acedward) [inline tools/smart-search/metadata.json:59]: This looks like it needs to be manually implemented, as the type was not extracted automatically from the type

File: tools/smart-search/metadata.json [hunk 2]
```diff
@@ -0,0 +1,74 @@
+{
+  "author": "@@my_local_ai.arb-sep-shinkai",
```
Comment (acedward) [inline tools/smart-search/metadata.json:2]:
```suggestion
  "author": "Shinkai",
```

File: tools/smart-search/tool.ts [hunk 3]
```diff
@@ -0,0 +1,443 @@
+import { getAccessToken } from './shinkai-local-support.ts';
```
Comment (acedward) [inline tools/smart-search/tool.ts:1]:
```suggestion
```

**Comment by guillevalin** [comment]
If @acedward is fine with this we could merge so we can start testing more advanced tools with the script im building.

**Review by guillevalin — APPROVED** [review]

**Pull request merged**